_rules/SC2-4-6-descriptive-labels.md

```diff
@@ -0,0 +1,125 @@
+---
+name: Label is descriptive
+test_type: atomic
+
+description: |
+  This rule checks that labels describe the purpose of form field elements.
+
+success_criterion:
+- 2.4.6 # Headings and labels
+
+test_aspects:
+- DOM Tree
+- CSS Styling
+
+authors:
+- Dagfinn Rømen
+- Geir Sindre Fossøy
+---
+
+## Test Procedure
+
+### Applicability
+
+This rule applies to any `label` element or HTML element referenced by aria-labelledby that is 
+- [visible](#visible) and included in the accessibility tree.
+- associated with an HTML element that has one of the listed form field [semantic roles](#semantic-role): `checkbox`, `combobox` (`select` elements), `listbox`, `menuitemcheckbox`, `menuitemradio`, `radio`, `searchbox`, `slider`, `spinbutton`, `switch` and `textbox`.
+
+**Note**: The list of form field roles is derived by taking all the [ARIA](https://www.w3.org/TR/wai-aria-1.1/) 1.1 roles that:
```

Comment:
Put the "1.1" into the link text -> [WAI-ARIA 1.1](https://www.w3.org/TR/wai-aria-1.1/)

Reply:
Updated. This text has been copy-pasted from https://github.com/auto-wcag/auto-wcag/blob/master/_rules/SC3-3-2-form-field-has-name.md and should probably be updated there as well.